nixos/modules/hardware/video/high-dpi.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, pkgs, config, ...}:
+with lib;
+
+{
+  options = {
+    hardware.video.high-dpi.enable = mkEnableOption "Font/DPI configuration optimized for High-DPI displays";
+  };
+  config = mkIf config.hardware.video.high-dpi.enable {
+    console.font = mkDefault
+      "${pkgs.terminus_font}/share/consolefonts/ter-u28n.psf.gz";
+
+    # Needed when typing in passwords for full disk encryption
+    console.earlySetup = mkDefault true;
+    boot.loader.systemd-boot.consoleMode = mkDefault "max";
+
+    services.xserver.dpi = mkDefault 196;
+    fonts.fontconfig.dpi = mkDefault 196;
+    environment.variables = mkDefault {
```

Comment:
Would it make sense to also set a variable for QT: `QT_AUTO_SCREEN_SCALE_FACTOR=1` ?

Reply:
As far as I know QT now automatically detects the correct DPI.